lore/encoders.py

```diff
@@ -564,6 +563,73 @@ def cardinality(self):
         return len(self.map) + 3
 
 
+class OneHot(Base):
+    """
+    Performs one hot encoding
+    """
+    def fit(self, data):
+        with timer(('fit one-hot %s:' % self.name), logging.DEBUG):
+            self.dummy_columns = self.get_dummies(data).columns.values
+            self.sequence_length = len(self.dummy_columns)
+
+    def get_dummies(self, data):
+        data = self.series(data)
+        return pandas.get_dummies(data, prefix=self.column)
+
+    def transform(self, data):
+        with timer('transform one_hot %s:' % self.name, logging.DEBUG):
+            dummies = self.get_dummies(data)
+            for col in [c for c in self.dummy_columns if c not in dummies.columns]:
+                dummies[col] = 0
+            return dummies[self.dummy_columns]
+
+    def get_column(self, encoded, i):
+        dummy = self.dummy_columns[i]
+        if dummy in encoded:
+            return encoded[dummy]
+        else:
+            return pandas.Series([0] * len(encoded))
+
+    def reverse_transform(self, data): pass
+
+    def cardinality(self):
+        return self.sequence_length
+
+    def sequence_name(self, i, suffix=''):
+        return (self.name + '_%i' + suffix) % i
+
+
+class CompressedOneHot(Unique):
```

Comment:
What do you think about combining `OneHot` as optional output on `Unique`, e.g. if you pass  `Unique(one_hot=True)`. My thinking is that it's better to have 1 class with various options, than 3. i.e. `CompressedOneHot` becomes `Unique(one_hot=True, minimum_occurences=10)`

Reply:
Personally, I like classes that do a few things and do them well. When the same class does too many things, as a user, it becomes difficult to understand how to get it to do, exactly what I want. Not to mention, that making sure all the options play well with each other is a bit of a pain to the developer as well. My preference would be to instead combine OneHot and CompressedOneHot. That's something I should have already done in the first place.